doc/Doxyfile

```diff
@@ -1034,25 +1034,6 @@ USE_HTAGS              = NO
 
 VERBATIM_HEADERS       = NO
 
-# If the CLANG_ASSISTED_PARSING tag is set to YES, then doxygen will use the
```

Comment:
Why to remove this?

Reply:
Before this update my version of Doxygen would print the following error message:

```
warning: Tag `CLANG_ASSISTED_PARSING' at line 1046 of file `build/doc/Doxyfile' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
warning: Tag `CLANG_OPTIONS' at line 1054 of file `build/doc/Doxyfile' belongs to an option that was not enabled at compile time.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u", or recompile doxygen with this feature enabled.
```

. Since we do not use these options at all, I do not think it makes any difference, if we just remove them from the `Doxyfile`.

Reply:
Activating these options might bring an benefit for the documentation of the C++ bindings (if Doxygen is compiled with clang support). But it is a minor thing, so it is okay to remove them so that these warnings are gone.